addons/website_slides/controllers/main.py

```diff
@@ -817,6 +819,7 @@ def slide_quiz_submit(self, slide_id, answer_ids):
         return {
             'goodAnswers': user_good_answers.ids,
             'badAnswers': user_bad_answers.ids,
+            'answersComments': dict((str(answer.question_id.id), answer.comment) for answer in user_answers),
```

Comment:
I think it's time to refactor this section a bit.
```
'answers': {
   answer.question_id.id: {'is_correct': answer.is_correct, 'comment': answer.comment}
   for answer in user_answers
}
```
It will allow you, on the JS side, to merge `_renderAnswersHighlighting` and `_renderAnswersComments` in a single method/loop and will be easier to read.
(In addition, it will be easier to add more information later if needed).